src/Api/WebhookApi.php

```diff
@@ -4,21 +4,68 @@
 
 namespace Sylius\PayPalPlugin\Api;
 
-use GuzzleHttp\ClientInterface;
+use GuzzleHttp\ClientInterface as GuzzleClientInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
 
 final class WebhookApi implements WebhookApiInterface
 {
-    private ClientInterface $client;
+    public function __construct(
+        private readonly GuzzleClientInterface|ClientInterface $client,
+        private readonly string $baseUrl,
+        private readonly ?RequestFactoryInterface $requestFactory = null,
+        private readonly ?StreamFactoryInterface $streamFactory = null,
+    ) {
+        if ($this->client instanceof GuzzleClientInterface) {
+            trigger_deprecation(
+                'sylius/paypal-plugin',
+                '1.6',
+                'Passing GuzzleHttp\ClientInterface as a first argument in the constructor is deprecated and will be removed. Use Psr\Http\Client\ClientInterface instead.',
+                self::class,
+            );
+        }
```

Comment:
We should probably mention this deprecations in the upgrade file 🤔

Reply:
I've added info to the UPGRADE.md